container/container_api.go

```diff
@@ -0,0 +1,27 @@
+package container
+
+import (
+	"io"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+)
+
+// ContainerAPI describes the API of container package.
+type ContainerAPI interface {
```

Comment:
I think it would be nice to have this renamed in `Container` and for the concrete implementation use `DockerContainer` for example, what do you think ?

Reply:
I don't understand the part related with _DockerContainer_. Could you explain more? If we rename _ContainerAPI_ to _Container_, we need to change current _Container_ type with another name. But I think we should keep that one as it is because package's name is _container_ and _Container_ is the main type of this package.

Reply:
The idea is to have the container package that has the `Container` interface so just the API. After that we can provide some concrete implementations, here for example Docker but we could have something different and so Docker is the actual implementation.

The optimal solution would be to have the following structure:
- container/
  - container.go
  - docker/
    - start.go
    - stop.go
    - ...

but for now having the interface and the implementation at the same level is fine.

So yes the actual `Container` should be `DockerContainer` or `Docker` or `ContainerDocker` something like that and the interface should be `Container`, that's the interface of the package

Reply:
Totally agree with @antho1404

Reply:
Totally agree with @antho1404.
It makes more sense to have the nicer name for the interface.

But let's keep the same file hierarchy for now: everything in the container folder.

Reply:
I'm ok to update names but I'm not sure about having multiple implementations of container like _docker_ or _kubernetes_ because the API may vary between those and we may end up with totally different _Container_ APIs.

We have the idea of refactoring _container package_ to collect more logic inside but even then I'm not sure that we'll have the same _Container_ API for different implementations.

Please lighten me if I'm missing something and this is an another topic, we can discuss it in forum.

Reply:
The multiple implementation of the container interface is just a future idea and will not be done now.
I will explain it more in post on the forum.